### PR TITLE
Add OTel Trace Source

### PR DESCRIPTION
**Pull request by kowshikn**
*Issue #55 *

*Description of changes:*
* Create a new Source plugin called ```otel_trace_source```
* This source is based of the new framework called [Armeria](https://github.com/line/armeria). This uses the OTLP protocol and supports OTLP/gRPC, OTLP/HTTP (protobuf) and OTLP/HTTP+JSON (partially).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: situp-plugins/apmtracesource/src/test/java/com/amazon/situp/plugins/source/oteltrace/OTelTraceSourceTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Test
+    void testHttpFullJson() throws InvalidProtocolBufferException, ExecutionException, InterruptedException {
+        addOneRequest();
+        final AggregatedHttpResponse res2 = WebClient.of().execute(RequestHeaders.builder()
+                        .scheme(SessionProtocol.HTTP)
+                        .authority("0.0.0.0:21890")
+                        .method(HttpMethod.POST)
+                        .path("/opentelemetry.proto.collector.trace.v1.TraceService/Export")
```
Comment (kowshikn) [inline situp-plugins/apmtracesource/src/test/java/com/amazon/situp/plugins/source/oteltrace/OTelTraceSourceTest.java:83]: We can have custom name for http after this [issue](https://github.com/line/armeria/issues/3072) is closed.

File: situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                sb.http(oTelTraceSourceConfig.getPort());
+            }
+            //TODO: Expose BlockingTaskExecutor/Connections
+            server = sb.build();
+        }
+        try {
+            server.start().get();
+            LOG.info("Started otel_trace_source...");
+        } catch (ExecutionException ex) {
+            LOG.error("otel_trace_source failed to start due to:", ex.getCause());
```
Comment (laneholloway) [inline situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java:54]: Were we going to throw an exception here?

File: situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        } catch (InterruptedException ex) {
+            LOG.error("otel_trace_source start got interrupted:", ex);
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(ex);
+        }
+    }
+
+    @Override
+    public void stop() {
+        if (server != null) {
```
Comment (laneholloway) [inline situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java:64]: Did we want to log if someone attempts to stop a non-started server for debugging purposes?

**Review by laneholloway — APPROVED** [review]
Please add java docs to the public APIs that we expect customers to use.

File: situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java (hunk 2 above)
Comment (erosas) [reply situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java:54]: +1 this looks like maybe a system.exit ?

File: situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java (hunk 3 above)
Comment (erosas) [reply situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java:64]: I would put it as a warn level

File: situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            try {
+                server.stop().get();
+                LOG.info("Stopped otel_trace_source.");
+            } catch (ExecutionException ex) {
+                LOG.error("otel_trace_source failed to stop due to:", ex.getCause());
```
Comment (erosas) [inline situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java:69]: why not just pass the whole exception? Are you sure the cause will be filled out?

File: situp-plugins/apmtracesource/src/test/java/com/amazon/situp/plugins/source/oteltrace/OTelTraceSourceTest.java (hunk 1 above)
Comment (erosas) [reply situp-plugins/apmtracesource/src/test/java/com/amazon/situp/plugins/source/oteltrace/OTelTraceSourceTest.java:83]: Nice

**Review by erosas — APPROVED** [review]
Minor comments, no blockers.

File: situp-plugins/apmtracesource/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# OTel Trace Source 
+
+This is a source which follows the [OTLP Protocol](https://github.com/open-telemetry/oteps/blob/master/text/0035-opentelemetry-protocol.md). This source supports ```OTLP/grpc``` and ```OTLP/HTTP```. Support for ```OTLP/HTTP+JSON``` is not complete due as the traceId and spanId will be ```base64``` and not ```HexString```.
```
[the remark this answers is not in the document]
Comment (kowshikn) [reply situp-plugins/apmtracesource/README.md:3]: Actually, this should be handled in the Armeria framework. Armenia uses a Jackson based protobuf, we need to add knob to convert HexString to byte array for traceId/spanId/parentSpanId. For now HTTP Json is not fully supported but in the future we can fix this.

File: situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java (hunk 2 above)
Comment (kowshikn) [reply situp-plugins/apmtracesource/src/main/java/com/amazon/situp/plugins/source/oteltracesource/OTelTraceSource.java:54]: I will check with the core pipeline and throw the cause as runtime.

**Pull request merged**